File: Python/Punto_2.py
from functools import reduce
# Cantidad de productos que comprará el cliente
n=int(input("Ingrese la cantidad de productos que va a comprar: "))
# Lista de precios de productos del cliente
precios=list(map(lambda elemento:int(input(f"Ingrese el precio del producto {elemento+1}: ")),range(n)))
# Se calcula el total a pagar sin el descuento, sumando cada precio
compraTotal=reduce(lambda acumulador=0,elemento=0:acumulador+elemento,precios)
# Se calcula el valor a pagar con un descuento de 15%
ventaDescuento=compraTotal*0.85
# Se imprimen el valor a pagar sin descuento y con el descuento
print(f"Valor a pagar sin descuento: {compraTotal}")
print(f"Valor a pagar con descuento: {ventaDescuento}")

File: Java/Punto2.java
package Java;

import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@FunctionalInterface
interface IngresoCompras {
    List<Integer> obtenerCompras();
}

public class Punto2 {
    public static void main(String[] args) {
        // Implementación de la interfaz funcional usando Scanner
        IngresoCompras ingresoCompras = () -> {
            Scanner scanner = new Scanner(System.in);
            System.out.println("Ingrese los montos de compras separados por espacios:");
            String input = scanner.nextLine();
            scanner.close(); // Para no gastar tanta memoria.
            return Stream.of(input.split(" "))
                    .map(Integer::parseInt)
                    .collect(Collectors.toList());
        };

        List<Integer> Vcompras = ingresoCompras.obtenerCompras();
        double compratotal = Vcompras.stream()
                .mapToDouble(V -> V) // Convertimos cada compra a double para usar sum()
                .sum();
        double ventaDescuento = compratotal * 0.85;
        System.out.println("Valor a pagar sin descuento: " + compratotal);
        System.out.println("Valor real a pagar con descuento: " + ventaDescuento);
    }
}